File: Backend/app/Models/PagoCxc.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class PagoCxc extends Model
{
    use HasFactory;

    protected $table = 'pagos_cxc';

    protected $fillable = [
        'cxc_id',
        'monto_pagado',
        'fecha_pago',
        'metodo_pago',
        'referencia',
        'cuenta_destino_id',
        'usuario_id',
        'asiento_id'
    ];

    protected $casts = [
        'fecha_pago' => 'date',
        'monto_pagado' => 'decimal:2',
    ];

    public function cuentaPorCobrar()
    {
        return $this->belongsTo(CuentaPorCobrar::class, 'cxc_id');
    }

    public function cuentaDestino()
    {
        return $this->belongsTo(CatalogoCuenta::class, 'cuenta_destino_id');
    }

    public function usuario()
    {
        return $this->belongsTo(User::class, 'usuario_id');
    }

    public function asientoContable()
    {
        return $this->belongsTo(AsientoContable::class, 'asiento_id');
    }
}

File: Frontend/lib/modulo_cxc/models/pago_cxc.dart
class PagoCxc {
  final int id;
  final int cxcId;
  final double montoPagado;
  final String fechaPago;
  final String metodoPago;
  final String? referencia;
  final Map<String, dynamic>? usuario;
  final Map<String, dynamic>? cuentaDestino;
  final Map<String, dynamic>? cuentaPorCobrar;

  PagoCxc({
    required this.id,
    required this.cxcId,
    required this.montoPagado,
    required this.fechaPago,
    required this.metodoPago,
    this.referencia,
    this.usuario,
    this.cuentaDestino,
    this.cuentaPorCobrar,
  });

  factory PagoCxc.fromJson(Map<String, dynamic> json) {
    return PagoCxc(
      id: json['id'],
      cxcId: json['cxc_id'],
      montoPagado: double.tryParse(json['monto_pagado']?.toString() ?? '0') ?? 0,
      fechaPago: json['fecha_pago'] ?? '',
      metodoPago: json['metodo_pago'] ?? 'Desconocido',
      referencia: json['referencia'],
      usuario: json['usuario'],
      cuentaDestino: json['cuenta_destino'],
      cuentaPorCobrar: json['cuenta_por_cobrar'],
    );
  }
}
